### PR TITLE
Add basic introspection_event method to the Runtime

### PR DESCRIPTION
**Pull request by juliettepretot**
# Checklist

See #913 for background

- [X] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [X] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/server/rust/oak_runtime/src/runtime/introspection_events.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  #[cfg(feature = "oak_debug")]
+  pub fn introspection_event(&self, event_details: EventDetails) {
+    let duration_since_unix_epoch = SystemTime::now()
+      .duration_since(UNIX_EPOCH)
+      .expect("Time went backwards");
+
+    let timestamp = Timestamp {
+      seconds: duration_since_unix_epoch.as_secs() as i64,
+      nanos: duration_since_unix_epoch.subsec_nanos() as i32,
+    };
```
Comment (daviddrysdale) [inline oak/server/rust/oak_runtime/src/runtime/introspection_events.rs:36]: nit: be nice to put this in a function of its own, so you can just have
```Rust
let event = Event {
  timestamp: Some(timestamp_for_now()),
  event_details: Some(event_details),
};
```

File: oak/server/rust/oak_runtime/src/runtime/introspection_events.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+pub use crate::proto::oak::introspection_events::event::EventDetails;
+use crate::{proto::oak::introspection_events::Event, runtime::Runtime};
+use log::info;
+use prost_types::Timestamp;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+// Introspection event related methods for the Runtime.
+impl Runtime {
+  /// Generates an introspection event recording a modification to the Oak
+  /// application.
```
Comment (daviddrysdale) [inline oak/server/rust/oak_runtime/src/runtime/introspection_events.rs:26]: nit: I'd word this as something like "... a modification to the Runtime's internal data structures" &ndash; there can be modifications to the Application (e.g. allocating more Wasm linear memory) that don't correspond to events or changes in the Runtime's structures.

File: oak/server/rust/oak_runtime/src/runtime/introspection_events.rs (hunk 1 above)
Comment (juliettepretot) [reply oak/server/rust/oak_runtime/src/runtime/introspection_events.rs:38]: Good idea :)

File: oak/server/rust/oak_runtime/src/runtime/introspection_events.rs (hunk 2 above)
Comment (juliettepretot) [reply oak/server/rust/oak_runtime/src/runtime/introspection_events.rs:28]: That's a good suggestion, I wasn't 100% happy with my phrasing too

**Comment by juliettepretot** [comment]
@rbehjati Good question, trying to debug currently, but having trouble reproducing it locally. 

https://pantheon.corp.google.com/cloud-build/builds/52082adb-859e-4245-ab63-3ef8156773cd;step=5?project=oak-ci

**Comment by juliettepretot** [comment]
Failing line in Cloud build:
image


Same line locally via `./scripts/docker_run ./scripts/runner run-ci`
image


🤔🤔🤔

**Comment by juliettepretot** [comment]
@rbehjati Found the reason after @tiziano88 helped me debug! It was two steps involving bazel running parallel, and it happening in this PR was likely timing only. 

See the slack thread for background: https://project-oak.slack.com/archives/CHE9E13C3/p1594984760219500

**Pull request merged**